### PR TITLE
SYNCT-182: Added feed filters configuration

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java [hunk 1]
```diff
@@ -1,76 +1,96 @@
...
-        feedConfigurationByOpenMrsClass = byOpenMrsClass;
-    }
+	private List<String> feedFilters;
+
+	private HashMap<String, FeedConfiguration> feedConfigurationByCategory;
+
+	private HashMap<String, FeedConfiguration> feedConfigurationByOpenMrsClass;
+
+	public FeedConfigurationServiceImpl() {
+		GeneralConfiguration generalConfiguration = null;
```
Comment (oskarhinc) [inline api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java]: no need to assign null to it I think

File: api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java [hunk 2]
```diff
@@ -1,76 +1,96 @@
...
+		AtomfeedUtils.writeFeedConfigurationToJsonFile(generalConfiguration,
+				ATOMFEED_PATH_TO_CUSTOM_CONFIGURATION);
+		loadFeedConfigurations(generalConfiguration);
+	}
+
+	@Override
+	public void saveConfig(String value) {
+		if (AtomfeedUtils.isValidateJson(value)) {
+			GeneralConfiguration localConfiguration = AtomfeedUtils.parseJsonStringToFeedConfiguration(value);
+			AtomfeedUtils.writeFeedConfigurationToJsonFile(localConfiguration,
```
Comment (oskarhinc) [inline api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java]: or just saveConfig(...)

File: api/src/test/resources/sampleFeedConfiguration2.json [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
-  "title" : "Title2",
-  "category": "patient",
-  "linkTemplates" : {
-    "rest" : "openmrs/ws/rest/v1/patient{uuid}?v=full",
-    "fhir" : "openmrs/ws/fhir/v1/patient{uuid}?v=full"
-  },
-  "feedWriter" : "custom"
-} ]
\ No newline at end of file
+{
```
Comment (oskarhinc) [inline api/src/test/resources/sampleFeedConfiguration2.json]: Maybe rename it to defaultBeanConfiguration, and the other one to custom...

File: api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java]: True

File: api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java (hunk 2 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/atomfeed/api/impl/FeedConfigurationServiceImpl.java]: Nice, thanks

File: api/src/test/resources/sampleFeedConfiguration2.json (hunk 3 above)
Comment (areklalo) [reply api/src/test/resources/sampleFeedConfiguration2.json]: Good idea, thanks

**Pull request merged**